lessons/models.py

```diff
@@ -81,13 +81,34 @@ def translate_proxy(self):
         return True
 
 
+"""
+Filterable
+
+"""
+
+
+class Filterable(Ownable):
+    class Meta:
+        abstract = True
+
+    # Helper method which should return true if the object is either owned by
+    # the user, or the user has permission to access all objects of this type.
+    def can_access(self, request):
+        raise NotImplementedError
+
+    # Restrict in-line editing of the object, overriding behavior defined here:
+    # http://mezzanine.jupo.org/docs/_modules/mezzanine/core/models.html#Ownable
+    def is_editable(self, request):
+        raise NotImplementedError
```

Comment:
should these return a reasonable default instead of erroring out? In particular, I can see value in doing something like 

```python
class Filterable
    # Helper method which should return true if the object is either owned by
    # the user, or the user has permission to access all objects of this type.
    def can_access(self, request):
        return request.user.id == self.user_id

    # Restrict in-line editing of the object, overriding behavior defined here:
    # http://mezzanine.jupo.org/docs/_modules/mezzanine/core/models.html#Ownable
    def is_editable(self, request):
        return self.can_access(request)
```

And then in subclasses doing

```python
class Vocab(Internationalizable, Filterable):
    def can_access(self, request):
        return request.user.has_perm('lessons.access_all_vocab') or super(Vocab, self).can_access(request)

    def is_editable(self, request):
        return request.user.has_perm('lessons.change_vocab') and super(Vocab, self).is_editable(request)
```

Reply:
I do not think we would want to `return self.can_access(request)` by default because this could unintentionally grant editing permission to someone without `change_*` permissions for a new object type. That might not be terrible, but it would violate the permission rules we've set up.

The idea was for this to be an abstract method which the subclass is required to define. If the subclass fails to implement it, that's a programming error so we raise. I'm not sure if this is the issue but I could document that in the comments if that helps.